src/parser/alerts.ts

```diff
@@ -0,0 +1,130 @@
+ */
+
+import MarkdownIt from 'markdown-it';
+import { config, configBaseDir } from '../config.js';
+import octicons from '@primer/octicons';
+import { existsSync, readFileSync } from 'fs';
+import { homedir } from 'os';
+import path from 'path';
+
+const githubAlertsIcons: Record<string, string> = {
+    note: 'info',
+    tip: 'light-bulb',
+    important: 'report',
+    warning: 'alert',
+    caution: 'stop',
+};
+const mergedIcons = {
+    ...githubAlertsIcons,
+    ...config.alertsOptions?.icons,
+};
+
+const fallbackIconOpt = config.alertsOptions?.fallbackIcon ?? mergedIcons['note'];
+const fallbackIcon = resolveIcon(fallbackIconOpt);
+const resolvedIcons: Record<string, string> = {};
+const titles = config.alertsOptions?.titles ?? {};
+
+for (const marker in mergedIcons) {
+    resolvedIcons[marker] = resolveIcon(mergedIcons[marker]);
+}
+
+function warnAndFallback(message: string): string {
+    return `<script>console.warn("${message}");</script>${fallbackIcon}`;
+}
+
+// Resolve option from alertsOptions.icons into raw svg tag
+function resolveIcon(iconOpt: string): string {
+    // Case 1: already a raw svg tag
+    if (iconOpt.startsWith('<svg')) {
+        return iconOpt;
+    }
+
+    // Case 2: svg file path
+    const prefix = ['/', './', '../', '~/'].find((p) => iconOpt.startsWith(p));
+    if (prefix && iconOpt.endsWith('.svg')) {
+        let iconPath = iconOpt;
+
+        if (prefix === '~/') {
+            iconPath = path.join(homedir(), iconPath.slice(2));
+        } else if (prefix === './' || prefix === '../') {
+            if (!configBaseDir) {
+                return warnAndFallback(`configBaseDir not set for relative icon path: ${iconPath}`);
+            }
+            iconPath = path.join(configBaseDir, iconPath);
+        }
+
+        if (!existsSync(iconPath)) {
+            return warnAndFallback(`Icon file not found: ${iconPath}`);
+        }
+        return readFileSync(iconPath).toString();
+    }
+
+    // Case 3: octicon name (in kebab-case) <https://primer.style/octicons>
+    const octiconName = iconOpt as keyof typeof octicons;
+    const octicon = octicons[octiconName]?.toSVG();
+
+    if (!octicon) {
+        return warnAndFallback(`Not a known octicon name: ${iconOpt}`);
+    }
+    return octicon;
+}
+
+function capitalize(str: string): string {
+    return str.charAt(0).toUpperCase() + str.slice(1);
+}
+
+const MarkdownItAlerts = (md: MarkdownIt) => {
+    // Allow multi word alphanumeric markers (includes underscore)
+    // Additionally allow dashes in marker
+    const markerRE = '[\\w- ]+';
+
+    // Match marker case insensitively
+    // Note: config icons and titles keys must be fully lowercase
+    const RE = new RegExp(`^\\\\?\\[\\!(${markerRE})\\]([^\\n\\r]*)`, 'i');
+
+    md.core.ruler.after('block', 'alerts', (state) => {
+        const tokens = state.tokens;
+        for (let i = 0; i < tokens.length; i++) {
+            if (tokens[i].type === 'blockquote_open') {
```

Comment:
Prefer negative case & continue, i.e.

```suggestion
            if (tokens[i].type !== 'blockquote_open') continue;
```

So that we can have the rest be one indent level less

Reply:
Addressed by [cdafc3d](https://github.com/jannis-baum/Vivify/pull/233/commits/cdafc3d4ba6003366a628ad13de652583e83e559)